app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt

```diff
@@ -44,36 +41,46 @@ class InstalledAddonDetailsFragment : Fragment() {
             addon = AddonDetailsFragmentArgs.fromBundle(requireNotNull(arguments)).addon
         }
 
-        return inflater.inflate(R.layout.fragment_installed_add_on_details, container, false).also {
-            bind(it)
-        }
+        return inflater.inflate(R.layout.fragment_installed_add_on_details, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        bindAddon(view)
     }
 
-    override fun onStart() {
-        super.onStart()
-        scope = requireContext().components.core.store.flowScoped { flow ->
-            flow.ifChanged { it.extensions }
-                .map { it.extensions.filterValues { extension -> extension.id == addon.id } }
-                .ifChanged()
-                .collect {
-                    val addonState = it[addon.id]
-                    if (addonState != null && addonState.enabled != addon.isEnabled()) {
-                        view?.let { view ->
-                            val newState = addon.installedState?.copy(enabled = addonState.enabled)
-                            this.addon = addon.copy(installedState = newState)
-                            view.enable_switch.setState(addon.isEnabled())
+    private fun bindAddon(view: View) {
+        lifecycleScope.launch(Dispatchers.IO) {
+            try {
+                val addons = requireContext().components.addonManager.getAddons()
+                lifecycleScope.launch(Dispatchers.Main) {
+                    runIfFragmentIsAttached {
+                        addons.find { addon.id == it.id }.let {
+                            if (it == null) {
+                                throw AddonManagerException(NullPointerException())
```

Comment:
nit: What about instead of the `NullPointerException`, we could create a `AddonManagerException("Addon ${it.id} not found")` indicating that we couldn't find the addon.

Reply:
@Amejia481 `class AddonManagerException(throwable: Throwable) : Exception(throwable)`, is there a way to pass string input instead?